src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -229,6 +229,9 @@ public bool IsExecutableStatement([NotNullWhen(true)] SyntaxNode? node)
         public bool IsGlobalStatement([NotNullWhen(true)] SyntaxNode? node)
            => node is GlobalStatementSyntax;
 
+        public SyntaxNode ExtractGlobalStatementOrSelf(SyntaxNode node)
+            => node is GlobalStatementSyntax globalStatement ? globalStatement.Statement : node;
```

Comment:
```suggestion
        public SyntaxNode GetStatementOfGlobalStatement(SyntaxNode node)
            => ((GlobalStatementSyntax)node).Statement;
```

Reply:
I "designed" it to be conditional and not break when the input node is not a global statement. While working with Roslyn I found that such functionality is useful when working with top-level code, but till this PR I managed to accomplish my goals without it. Reworking it the way you suggest will require checking for global statement almost each time, because usually there are both cases for global statement and at the same time for another non-global context. Even in the context of this PR changing the method without additional refactoring will break normal block-based scenarios. Considering everithing above, do you still want this change to be made?

Reply:
@CyrusNajmabadi Will it be ok, if I do `TryGetStatementOfGlobalStatement`?